workorder-job-repeater.lua

```diff
@@ -0,0 +1,278 @@
+
+    -- for debug purposes: pause the game
+    --df.global.pause_state = true
+
+    log(DEBUG, "Repeat job " .. dfhack.job.getName(job) .. " (id " .. job.id .. ") from order " .. order.id)
+
+    -- we need a clone because original(which itself is a copy) is destroyed or something.
+    -- cloneJobStruct doesn't copy flags over.
+    local by_manager = job.flags.by_manager
+    local do_now = job.flags.do_now
+    job = dfhack.job.cloneJobStruct(job)
+    job.flags.by_manager = by_manager
+    job.flags.do_now = do_now
+
+    -- step 0: adjust what we can right now
+    job.flags.working = false
+    job.completion_timer = -1
+
+    -- remove old items we worked upon
+    log(DEBUG, "Freeing old items")
+    for _, item in pairs(job.items) do
+        item:delete()
+    end
+    job.items:resize(0)
+
+    local workshop = nil
+    local ixs = {}
+    for ix, gref in ipairs(job.general_refs) do
+        if gref:getType() == df.general_ref_type.BUILDING_HOLDER then
+            local building = gref:getBuilding()
+            if building then
+                log(DEBUG, "Building from gref: " .. building.id)
+                workshop = building
+            else
+                -- should never ever happen, I think
+                log(WARN, "Building from gref not found: ", gref.building_id)
+            end
+        else
+            ixs[ix] = true
+        end
+    end
+    for ix, gref in pairs(ixs) do
+        log(DEBUG, "Removing gref ", gref)
+        gref:delete()
+        job.general_refs:erase(ix)
+    end
+    log(DEBUG, "#gref: " .. #job.general_refs .. " (should be 1)")
+
+    -- step 1: associate a posting with it
+    local ok = addJobToPostings(job)
+    if not ok then
+        log(ERROR, "Couldn't add job of order.id " .. order.id .. " to a posting")
+        return
+    end
+
+    -- step 2: add the job to the job list
+    local ok = dfhack.job.linkIntoWorld(job, false) -- we don't need a new id, I think?
+    if not ok then -- fallback just in case
+        log(DEBUG, "Couldn't add the job using job.id " .. job.id .. ". Requesting new id.")
+        dfhack.job.linkIntoWorld(job, true)
+    end
+    log(DEBUG, "Job " .. job.id .. " is linked.")
+
+    -- step 3: if a building is associated with the work order, find it
+    if order.workshop_id < 0 then
+        log(DEBUG, "This work order is not associated with a workshop: done.")
+        return
+    end
+
+    if not workshop then
+        log(WARN, "No workshop despite order.workshop_id " .. order.workshop_id .. " -- restoring")
+        workshop = findBuildingById(order.workshop_id)
+    end
+    log(DEBUG, "Associated workshop: " .. workshop.id)
+
+    -- step 4: add this job to the workshop's job list.
+    workshop.jobs:insert('#', job)
+    log(DEBUG, "Added job " .. job.id .. " to workshop.jobs")
+end
+
+frequency = frequency or 10
+local function start(N)
+    frequency = tonumber(N) or frequency
+    if eventful.onJobCompleted.workorder_repeat_job_immediately then
```

Comment:
this should use the name of the script -- `eventful.onJobCompleted.workorder_job_repeater` (or `eventful.onJobCompleted['workorder-job-repeater']`)